### PR TITLE
use our own systemd units and restart kubelet more often

### PR DESCRIPTION
**Pull request by BenTheElder**
xref: #576 

this can shave around 10 seconds off of startup with v1.14.2

for v1.15+ we still need to track down issues in kubernetes, but this does still generally make things faster, sub 30s startup for a single node is now pretty normal with this patch

we'll want to get new images with this at some point, I'll probably wait for more of the kindnetd work to land though

### CODE REVIEW

File: pkg/build/node/systemd.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Description=kubelet: The Kubernetes Node Agent
+Documentation=http://kubernetes.io/docs/
+
+[Service]
+ExecStart=/usr/bin/kubelet
+Restart=always
+StartLimitInterval=0
+# NOTE: kind deviates from upstream here with a lower RestartSec
+# 100ms is actually the systemd default, we'll use that for now
+RestartSec=100ms
```
Comment (BenTheElder) [inline pkg/build/node/systemd.go]: https://www.freedesktop.org/software/systemd/man/systemd.service.html

**Comment by BenTheElder** [comment]
... these changes override what the existing builds do so I updated the PR to:
- drop the `apt` "build" (we'd been talking about this for a while anyhow, and it doesn't fit with this)
- stop handling systemd units and related files in the "bits" implementations

**Comment by aojea** [comment]
> ... these changes override what the existing builds do so I updated the PR to:
> 
> * drop the `apt` "build" (we'd been talking about this for a while anyhow, and it doesn't fit with this)
> * stop handling systemd units and related files in the "bits" implementations

IIRC apt build was used by some specific architecture, I think that we should provide the tar option ASAP for people that don't want/need to compile k8s from source

**Comment by tao12345666333** [comment]
> I think that we should provide the tar option ASAP for people that don't want/need to compile k8s from source

+1

**Comment by aojea** [comment]
heheh, some of the tests run now in 18.30 min, usually they were running in about 20 mins

**Comment by BenTheElder** [comment]
> IIRC apt build was used by some specific architecture, I think that we should provide the tar option ASAP for people that don't want/need to compile k8s from source

we already provide images from all of the current releases, the apt build literally only allowed installing the current release. there's on reason to build that

we should however provide the tar option, but that's somewhere in the mix with all the other changes we need to get in :sweat_smile: 

> heheh, some of the tests run now in 18.30 min, usually they were running in about 20 mins

we'll see how stable that is, if you look it's affected quite a bit by other things (like, prow should not be seeing heavy I/O throttling, which was causing a lot of flakes, because Katharine and I allocated bigger disks in the build cluster :-))

**Pull request merged**